pkg/component/kubeapiserver/deployment.go

```diff
@@ -225,6 +225,12 @@ func (k *kubeAPIServer) reconcileDeployment(
 					RestartPolicy:                 corev1.RestartPolicyAlways,
 					SchedulerName:                 corev1.DefaultSchedulerName,
 					TerminationGracePeriodSeconds: pointer.Int64(30),
+					SecurityContext: &corev1.PodSecurityContext{
+						RunAsNonRoot: pointer.Bool(true),
+						RunAsUser:    pointer.Int64(65532),
+						RunAsGroup:   pointer.Int64(65532),
+						FSGroup:      pointer.Int64(65532),
```

Comment:
I would suggest that we add a comment so no one wonders where this user comes from. The same can be done for kube-scheduler and KCM.
```suggestion
                                                  // use the nonroot user from a distroless container
                                                  // https://github.com/GoogleContainerTools/distroless/blob/1a8918fcaa7313fd02ae08089a57a701faea999c/base/base.bzl#L8
						RunAsNonRoot: pointer.Bool(true),
						RunAsUser:    pointer.Int64(65532),
						RunAsGroup:   pointer.Int64(65532),
						FSGroup:      pointer.Int64(65532),
```